ui/src/dataLoaders/components/TelegrafEditorMonaco.tsx

```diff
@@ -0,0 +1,136 @@
+import React, {PureComponent} from 'react'
+import {connect} from 'react-redux'
+import {AppState} from 'src/types'
+import Editor from 'src/shared/components/TomlMonacoEditor'
+import {setText, setActivePlugins} from 'src/dataLoaders/actions/telegrafEditor'
+
+const PLUGIN_REGEX = /\[\[\s*(inputs|outputs|processors|aggregators)\.(.+)\s*\]\]/
+
+interface DispatchProps {
+  onSetText: typeof setText
+  onSetActivePlugins: typeof setActivePlugins
```

Comment:
I believe you mentioned that we're going to be setting types differently from hereon, or is this the way?

Reply:
no that's in the `action` pattern i think you're referring to